lib/puppet/provider/file_line/ruby.rb

```diff
@@ -83,6 +83,9 @@ def lines
   rescue TypeError => _e
     # Ruby 1.8 doesn't support open_args
     @lines ||= File.readlines(resource[:path])
+  rescue Errno::ENOENT => _e
+    raise _e unless resource.noop?
```

Comment:
I think this is the best way to reraise the exception:
```suggestion
  rescue Errno::ENOENT
    raise unless resource.noop?
```

Reply:
Change made.  Thanks for the review.